bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -56,8 +56,7 @@ def run_task(targets, task, args = {}, options = {})
 
   # Returns true if the target has the puppet-agent feature defined, either from inventory or transport.
   def agent?(target, executor, inventory)
-    inventory.features(target).include?('puppet-agent') ||
```

Comment:
We shouldn't remove this line - if users have the `puppet-agent` feature defined in their inventory for certain targets this will still attempt to install the agent, when it should skip them and not try to install the agent.